### PR TITLE
Handle Non-existent Expansion + Expose Logspam + Nonlethal kills with 1  max hp

### LINKED ISSUE #16: Using Artifact of Metamorphosis with custom survivors locked behind custom expansions causes a softlock
What the title says, Having any SurvivorDef that has an ExpansionDef causes a softlock when spawning, where the player ends up in limbo and requires a run restart. This only occurs if the ExpansionDef has no tied in EntitlementDef (AKA, not tied to any purchase)

This is an easy to reproduce bug, just install Starstorm 2 on a fresh savefile and play with the artifact enabled.

image
Seems like either ILHooking or replacing the method with an On. hook is necesary.
[LogOutput.log](https://github.com/risk-of-thunder/RoR2BepInExPack/files/11363371/LogOutput.log)

### PR DESCRIPTION
**Pull request by yekoc**
- ModCompat: Handle expansions without entitlement :  Fixes #16 
 Mods don't often have their own entitlement defs because creating them is a huge pain,but the game assumes they exist if expansions are involved.
 Fix: Everyone is entitled to having a Nothing.
- VanillaFixes: Remove Expose Logspam : 
  The game prints an unnecessary line whenever expose is applied via the damage type
  Fix: Don't.
- VanillaFixes: Fix NonLethal damage still killing when you have 1 max hp
  TakeDamage has some emergent sketchy float math that seems to fail without reason when the body has 1 max hp,causing the nonlethal damage check to fail
  Fix: Insert a different check,the way it ended up is not too clear on first sight because the vanilla IL is set up slightly inconviniently.

### CODE REVIEW

**Comment by Nebby1999** [comment]
Are you sure thast handle expansions without entitlement works properly' by the looks of it it seems like youre returning immediatly? that would skip the checks for SOTV expansionDef which would technically mean you dont need to purchase SOTV to get the expansiondef.


Seems a bit sketchy that bit

**Comment by yekoc** [comment]
that return is within the check vanilla already does
it effectively just turns the throw into return true

File: RoR2BepInExPack/VanillaFixes/FixNonLethalOneHP.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        bool ILFound = c.TryGotoNext( MoveType.After,
+                x => x.MatchLdfld(typeof(HealthComponent).GetField(nameof(HealthComponent.health),ReflectionHelper.AllFlags)),
+                x => x.MatchLdcR4(1),
+                x => x.MatchBltUn(out _));
+
+        if (ILFound)
+        {
+            c.Index--;
+            c.Emit(OpCodes.Ldarg_0);
+            c.EmitDelegate<Func<float,HealthComponent,float>>((orig,self) => (self.health > 0f) ? 0f : orig);
```
Comment (harbingerofme) [inline RoR2BepInExPack/VanillaFixes/FixNonLethalOneHP.cs]: `orig,self` are common descriptors of other types of variables in our modding scene, can we consider `newHealth, self` instead?

Comment (yekoc) [reply RoR2BepInExPack/VanillaFixes/FixNonLethalOneHP.cs]: I have been getting a bit used to using orig as 'original value' when emitting value replacement delegates of that form so it bled through unintentionally.
You are correct that it doesn't fit with the project/community convention,so it's better if I fix it,let me amend the commit real quick.

**Pull request merged**